server/planning/content_profiles/planning_types_async_service.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2021 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from typing import Any
+from copy import deepcopy
+
+from superdesk.core.resources import AsyncResourceService
+from superdesk.core.types import SearchRequest, ProjectedFieldArg
+from superdesk.utils import ListCursor
+
+from planning.common import planning_link_updates_to_coverage, get_config_event_related_item_search_provider_name
+from planning.types import PlanningTypesResourceModel
+from .profiles import DEFAULT_PROFILES
+
+
+class PlanningTypesAsyncService(AsyncResourceService[PlanningTypesResourceModel]):
+    """Planning types async service
+
+    Provide a service that returns what fields should be shown in the edit forms in planning, in the edit dictionary.
+    Also provide a schema to allow the client to validate the values entered in the forms.
+    Entries can be overridden by providing alternates in the planning_types mongo collection.
+    """
+
+    async def find_one(
+        self,
+        req: SearchRequest | None = None,
+        projection: ProjectedFieldArg | None = None,
+        use_mongo: bool = False,
+        version: int | None = None,
+        **lookup,
+    ) -> PlanningTypesResourceModel | None:
```

Comment:
Can you please add some docstrings here as to why we are overriding the `find_one` method?